core/src/replay_stage.rs

```diff
@@ -703,7 +703,10 @@ impl ReplayStage {
         last_entry: &Hash,
         shred_index: usize,
     ) -> Result<()> {
-        if !entries.verify(last_entry) {
+        let mut entry_state = entries.start_verify(last_entry);
+        blocktree_processor::process_entries(bank, entries, true)?;
+
+        if !entry_state.finish_verify(entries) {
```

Comment:
Later we can also make this not block replay stage. Do this entirely in the background and allow it to mark slots as dead. 
It's a little bit more involved(we need to delay voting till verify is done) so definitely not needed as part of this pr :)

Reply:
@sagar-solana you're saying move `finish_verify` to when we are about to vote on the slot? This gives us time to run the expensive bank weight calculation while the entries are still verifying. 

I think that might be helpful only if the verification is taking longer than the `process_entries()` call here, otherwise the entries will already have finished verification,  and you best check the results here to avoid incurring an expensive bank weight calculation if the entries didn't verify.

Reply:
No I meant that we should make `is_verified` a part of the slot's meta. That way the verification can happen entirely in the background and we'd only check if everything verified when we look at the slot meta before voting on a slot.

Reply:
We can do in a followup pr